rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/TechnologyIdentified.java

```diff
@@ -113,27 +110,20 @@ public int getCount()
     public void perform(GraphRewrite event, EvaluationContext context, WindupVertexFrame payload)
     {
         Set<ProjectModel> projects = new HashSet<>();
-        if (payload instanceof ProjectModel)
-        {
-            projects.add((ProjectModel) payload);
-        }
-        else if (payload instanceof FileModel)
-        {
-            projects.add(((FileModel) payload).getProjectModel());
-        } else if (payload instanceof FileLocationModel)
-        {
-            projects.add(((FileLocationModel)payload).getFile().getProjectModel());
-        } else if (payload instanceof JNDIResourceModel)
-        {
-            JNDIResourceModel jndiResourceModel = (JNDIResourceModel)payload;
-            jndiResourceModel.getApplications().forEach(projects::add);
-        } else if (payload instanceof EjbBeanBaseModel)
+
+        if (payload instanceof BelongsToProject)
         {
-            EjbBeanBaseModel ejbBeanBaseModel = (EjbBeanBaseModel) payload;
-            ejbBeanBaseModel.getApplications().forEach(projects::add);
+            Iterable<ProjectModel> rootProjectModels = ((BelongsToProject) payload).getRootProjectModels();
+
+            for (ProjectModel projectModel : rootProjectModels)
+            {
+                projects.add(projectModel);
+            }
```

Comment:
As discussed, you can replace it with 
```
((BelongsToProject) payload).getRootProjectModels().forEach(projects::add);
```

Reply:
You are right. Are there any performance drawbacks of this approach? Because this one is easier to debug, if you need to see which projects are added.

Reply:
well, for such simple case lambda is fine and readable enough I think. if you need to debug it that is the different thing and could be a little harder

Reply:
I'm still concerned that this is going to attach everything to the shared-libs project in the case of a shared jar.

Reply:
I wrote a unit test and confirmed that it returns only the shared-libs project in the case of a library in shared libs.

Updated PR here: https://github.com/windup/windup/pull/1178